Integrations/DuoAdminApi/DuoAdminApi.py

```diff
@@ -35,23 +37,93 @@
 }
 
 
-# Utility Methods
+def override_make_request(self, method, uri, body, headers):
+    """
+
+    This function is an override function to the original
+    duo_client.client.Client._make_request function in API version 4.1.0
+
+    The reason for it is that the API creates a bad uri address for the GET requests.
+
+    """
+
+    conn = self._connect()
+
+    # Ignored original code #
+    # --------------------- #
+    # if self.proxy_type == 'CONNECT':
+    #     # Ensure the request uses the correct protocol and Host.
+    #     if self.ca_certs == 'HTTP':
+    #         api_proto = 'http'
+    #     else:
+    #         api_proto = 'https'
+    #     uri = ''.join((api_proto, '://', self.host, uri))
+    # ------------------- #
+    # End of ignored code #
+
+    conn.request(method, uri, body, headers)
+    response = conn.getresponse()
+    data = response.read()
+    self._disconnect(conn)
+    return (response, data)
 
 
+# Utility Methods
+
 def create_api_call():
-    if USE_SSL:
-        return duo_client.Admin(
+    if USE_SSL and not USE_PROXY:
+        # Use of both is not working with the API
+        client = duo_client.Admin(
+            ikey=INTEGRATION_KEY,
+            skey=SECRET_KEY,
+            host=HOST,
+        )
+    else:
+        client = duo_client.Admin(
             ikey=INTEGRATION_KEY,
             skey=SECRET_KEY,
             host=HOST,
             ca_certs='DISABLE'
         )
 
-    return duo_client.Admin(
-        ikey=INTEGRATION_KEY,
-        skey=SECRET_KEY,
-        host=HOST
-    )
+    client._make_request = lambda method, uri, body, headers: override_make_request(client, method, uri, body, headers)
+
+    return client
+
+
+def set_proxy():
+    try:
+        proxy_settings = os.environ.get('HTTP_PROXY') or os.environ.get('http_proxy', '')
+        if proxy_settings:
+            host, port = get_host_port_from_proxy_settings(proxy_settings)
+
+            if USE_PROXY:
+                admin_api.set_proxy(host=host, port=port)
+                return
```

Comment:
redundant

Reply:
@yaakovi why?

Reply:
because you exit the function anyway after this line, you don't need this kind of returns in python